pkgs/regen.nix

```diff
@@ -0,0 +1,53 @@
+let
+  localLib = import ../lib.nix;
+  jemallocOverlay = self: super: {
+    # jemalloc has a bug that caused cardano-sl-db to fail to link (via
+    # rocksdb, which can use jemalloc).
+    # https://github.com/jemalloc/jemalloc/issues/937
+    # Using jemalloc 510 with the --disable-initial-exec-tls flag seems to
+    # fix it.
+    jemalloc = self.callPackage ../nix/jemalloc/jemalloc510.nix {};
+  };
+in
+{ system ? builtins.currentSystem
+, config ? {}
+, pkgs ? (import (localLib.fetchNixPkgs) { inherit system config; overlays = [ jemallocOverlay ]; })
+
+# Update this if you need a package version recently uploaded to hackage.
+# Any timestamp works.
+, hackageSnapshot ? "2018-07-17T09:58:14Z"
+}:
+
+with pkgs;
+
+let
+  iohkPkgs = import ../. { inherit system config pkgs; };
+  deps = [ nixStable coreutils git findutils cabal2nix glibcLocales stack cabal-install iohkPkgs.stack2nix ];
+in
+  writeScript "generate.sh" ''
+    #!${stdenv.shell}
+    set -euo pipefail
+    export PATH=${lib.makeBinPath deps}
+    export HOME=$(pwd)
+
+    echo "Using hackage snapshot from ${hackageSnapshot}"
+
+    # Ensure that nix 1.11 (used by cabal2nix) works in multi-user mode.
```

Comment:
if `nixStable` is in deps and deps comes from our pkgs, why do we care about nix 1 support? Or is this in the case someone is running a nix 1 daemon still (in which case shouldn't we encourage them to upgrade nix)? I'm fine with this as-is, but just wondering when we start to drop the nix 1 support from cardano-sl repo, especially for developer tools like `pkgs/generate.sh` where we can encourage people to upgrade.

Reply:
nix 1.11 is hard-coded into cabal2nix so I need to add `NIX_REMOTE` to get it to work when the user is running nix 2.